server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/MirrorRunner.java

```diff
@@ -95,50 +95,54 @@ public MirrorRunner(ProjectApiManager projectApiManager, CommandExecutor command
-                        }
-
-                        return mirrorAccessController.isAllowed(mirror).thenApplyAsync(allowed -> {
-                            if (!allowed) {
-                                throw new MirrorAccessException(
-                                        "The mirroring from " + mirror.remoteRepoUri() + " is not allowed: " +
-                                        mirrorKey.projectName + '/' + mirrorKey.mirrorId);
-                            }
-
-                            final String zone = mirror.zone();
-                            if (zone != null && !zone.equals(currentZone)) {
-                                throw new MirrorException(
-                                        "The mirror is not in the current zone: " + currentZone);
-                            }
-                            final MirrorTask mirrorTask = new MirrorTask(mirror, user, Instant.now(),
-                                                                         currentZone, false);
-                            final MirrorListener listener = MirrorSchedulingService.mirrorListener();
-                            listener.onStart(mirrorTask);
-                            try {
-                                final MirrorResult mirrorResult =
-                                        mirror.mirror(workDir, commandExecutor,
-                                                      mirrorConfig.maxNumFilesPerMirror(),
-                                                      mirrorConfig.maxNumBytesPerMirror(),
-                                                      mirrorTask.triggeredTime());
-                                listener.onComplete(mirrorTask, mirrorResult);
-                                return mirrorResult;
-                            } catch (Exception e) {
-                                listener.onError(mirrorTask, e);
-                                throw e;
-                            }
-                        }, worker);
-                    });
+                    metaRepo(mirrorKey.projectName).mirror(mirrorKey.repoName, mirrorKey.mirrorId).thenCompose(
+                            mirror -> {
+                                if (!mirror.enabled()) {
+                                    throw new MirrorException("The mirror is disabled: " +
+                                                              mirrorKey.repoName + '/' + mirrorKey.mirrorId);
+                                }
+
+                                return mirrorAccessController.isAllowed(mirror).thenApplyAsync(allowed -> {
+                                    if (!allowed) {
+                                        throw new MirrorAccessException(
+                                                "The mirroring from " + mirror.remoteRepoUri() +
+                                                " is not allowed: " +
+                                                mirrorKey.projectName + '/' + mirrorKey.mirrorId);
```

Comment:
To apply the same approach as above here
```suggestion
                                                mirrorKey.repoName + '/' + mirrorKey.mirrorId);
```